src/control/server/instance.go

```diff
@@ -190,8 +190,15 @@ func (srv *IOServerInstance) NeedsScmFormat() (bool, error) {
 		return false, err
 	}
 
-	srv.log.Debugf("%s (%s) needs format: %t", scmCfg.MountPoint, scmCfg.Class, !res.Formatted)
-	return !res.Formatted, nil
+	// Logic:
+	// 1. If the SCM is mounted, assume here that it doesn't need to be formatted.
+	// 2. If not mounted, check:
+	// 2a. Does it appear to be formatted with something?
+	// AND
+	// 2b. Does it appear to be mountable by us?
+	needsFormat := !res.Mounted && !(res.Formatted && res.Mountable)
```

Comment:
the fact that you need comments here is a sign that the logic should be split so so it's not so hard to reason about the line (as per solid guidelines)

Reply:
I can't see any sensible simpler presentation though unfortunately

Reply:
Thanks for pointing this out. After thinking about it some more I realized that the formatted state is irrelevant here. All we care about this point is whether or not the SCM is already mounted or is mountable without a format operation.